1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,34 @@
-function setAlarm() {}
+function setAlarm() {
+  let setAlarm = document.querySelector("#alarmSet")
+  const startingMinutes = setAlarm.value;
+  let time = startingMinutes * 60;
+
+  let timeRemaining = document.querySelector("#timeRemaining");
+
+  setInterval(countDown, 1000);
+
+  function countdown() {
+    const minutes = Math.floor(time / 60);
+    let seconds = time % 60;
+    seconds = seconds < 10 ? `0` + seconds : seconds;
+    timeRemaining.textContent = `Time Remaining: ${minutes}: ${seconds}`;
+    if (time == 0) {
+      timeRemaining.textContent = `Time Remaining: 00:00`;
+      playAlarm();
+      let colors = ["red", "green", "yellow", "purple", "pink"];
+      let index = 0;
+      function changeColor() {
+        document.querySelector(".centre").style.backgroundColor = colors[index];
+        index = (index + 1) % colors.length;
+        setTimeout(changeColor, 100);
+      }
+      changeColor();
+    } else if (time > 0) {
+      time--;
+    }
+  }
+
+}
```

Comment:
Ohh, so close with this one Mo! here are a few pointers I would recommend.

Whenever you are setting a DOM or HTML element in your JS script, you should do this on a global scope. That way all your functions have access to this. As they won't change on an element level, you should also set them as a const. ie:

const timeRemaining = document.querySelector("#timeRemaining");
const alarm = document.querySelector("#alarmSet");
Another tip is that you should always assign your setInterval to a variable. If you don't do this, your code cannot keep track of the interval, and therefore can never stop it. Another issue is that if you click "Set Alarm" over and over again, you are adding a new interval each time and the page very quickly starts breaking.

Another issue is you have declared your change color function inside where it is called. You should always make all your function on a top-level, and then call them when needed.

I refactored your code to this to got it to work. I hope this helps to look at!

const timeRemaining = document.querySelector("#timeRemaining");
const alarm = document.querySelector("#alarmSet");
let countdown;
let colorbg;

function setAlarm() {
	clearInterval(countdown);
	let startingTime = alarm.value;
	alarm.value = null;
	countdown = setInterval(()=> {
		console.log(startingTime);
		if(startingTime === 0) {
			timeRemaining.innerHTML = "Time Remaining: 00:00";
			playAlarm();
			colorbg = setInterval(bgColorChange, 1000);
		}
		else if(startingTime > 0) {
			startingTime --;
			timeRemaining.innerHTML = `Time Remaining: ${startingTime}`;
		}
	}, 1000);
}

const colors = ["red", "green", "yellow", "purple", "pink"];
const bgColorChange = ()=> {
	const index = Math.floor(Math.random() * 5);
	document.getElementById("centre").style.backgroundColor = colors[index];
};

// DO NOT EDIT BELOW HERE
var audio = new Audio("alarmsound.mp3");
function setup() {
	document.getElementById("set").addEventListener("click", () => {
		setAlarm();
	});
	document.getElementById("stop").addEventListener("click", () => {
		pauseAlarm();
	});
}
function playAlarm() {
	clearInterval(countdown);
	audio.play();
}
function pauseAlarm() {
	clearInterval(countdown);
	clearInterval(colorbg);
	alarm.value = null;
	timeRemaining.innerHTML = "Time Remaining: 00:00";
	document.getElementById("centre").style.backgroundColor = "white";
	audio.pause();
}
window.onload = setup;